launch_ros/launch_ros/descriptions/node.py

```diff
@@ -0,0 +1,327 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# DISTRIBUTION A. Approved for public release; distribution unlimited.
+# OPSEC #4584.
+#
+# Delivered to the U.S. Government with Unlimited Rights, as defined in DFARS
+# Part 252.227-7013 or 7014 (Feb 2014).
+#
+# This notice must appear in all copies of this file and its derivatives.
+
+"""Module for a description of a Node."""
+
+import os
+import pathlib
+
+from tempfile import NamedTemporaryFile
+
+from typing import Dict
+from typing import Iterable
+from typing import List
+from typing import Optional
+from typing import Text
+from typing import Tuple
+from typing import Union
+
+from launch import Action
+from launch import LaunchContext
+from launch import SomeSubstitutionsType
+from launch.descriptions import Executable
+import launch.logging
+from launch.substitutions import LocalSubstitution
+from launch.utilities import ensure_argument_type
+from launch.utilities import normalize_to_list_of_substitutions
+from launch.utilities import perform_substitutions
+
+from launch_ros.utilities import add_node_name
+from launch_ros.utilities import evaluate_parameters
+from launch_ros.utilities import get_node_name_count
+from launch_ros.utilities import make_namespace_absolute
+from launch_ros.utilities import normalize_parameters
+from launch_ros.utilities import normalize_remap_rules
+from launch_ros.utilities import prefix_namespace
+
+from rclpy.validate_namespace import validate_namespace
+from rclpy.validate_node_name import validate_node_name
+
+import yaml
+
+from .node_trait import NodeTrait
+from ..parameter_descriptions import Parameter
+from ..parameters_type import SomeParameters
+from ..remap_rule_type import SomeRemapRules
+
+
+class Node:
+    """Describes a ROS node."""
+
+    UNSPECIFIED_NODE_NAME = '<node_name_unspecified>'
+    UNSPECIFIED_NODE_NAMESPACE = '<node_namespace_unspecified>'
+
+    def __init__(
+        self, *,
+        node_name: Optional[SomeSubstitutionsType] = None,
+        node_namespace: Optional[SomeSubstitutionsType] = None,
+        parameters: Optional[SomeParameters] = None,
+        remappings: Optional[SomeRemapRules] = None,
+        traits: Optional[Iterable[NodeTrait]] = None,
+        **kwargs
+    ) -> None:
+        """
+        Construct an Node description.
+
+        Many arguments are passed eventually to
+        :class:`launch.actions.ExecuteProcess`, so see the documentation of
```

Comment:
@roger-strain did you mean to `launch.actions.ExecuteLocal`? At any rate, it shouldn't specify an executor. Perhaps documentation needs a revisit?

Reply:
Definitely was a copy/paste leftover from something. I've removed the clause entirely. I think the rest of it is okay, but if there are other deficiencies please let me know.